plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/util/CephContainer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.util;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.containers.wait.strategy.SelectedPortWaitStrategy;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+
+import java.io.Closeable;
+import java.time.Duration;
+
+import static java.lang.String.format;
+
+public final class CephContainer
+        implements Closeable
+{
+    public static final String CEPH_S3_ACCESS_KEY = "ceph-s3-access-key";
+    public static final String CEPH_S3_SECRET_KEY = "ceph-s3-secret-key";
+    public static final int CEPH_S3_PORT = 8000;
+
+    private final DockerContainer ceph;
+
+    public CephContainer(Network network)
+    {
+        ceph = new DockerContainer("ceph/daemon:latest-master")
+                .withEnv(ImmutableMap.<String, String>builder()
+                        .put("MON_IP", "127.0.0.1")
+                        .put("OSD_COUNT", "1")
+                        .put("CLUSTER", "ceph")
+                        .put("CEPH_PUBLIC_NETWORK", "0.0.0.0/0")
+                        .put("CEPH_DEMO_UID", "ceph")
+                        .put("CEPH_DEMO_ACCESS_KEY", CEPH_S3_ACCESS_KEY)
+                        .put("CEPH_DEMO_SECRET_KEY", CEPH_S3_SECRET_KEY)
+                        .put("RGW_NAME", "ceph")
+                        .put("RGW_FRONTEND_PORT", String.valueOf(CEPH_S3_PORT))
```

Comment:
A Ceph demo cluster is used to run a single Docker container with all required Ceph services. The demo cluster is used in Ceph tests as well (https://github.com/ceph/ceph-container/blob/main/.github/workflows/container.yml#L14).

A `latest-master` image is used because as of August 2021, new Ceph container images are pushed to the quay.io registry only. Docker hub won't receive new content for that specific image but current images remain available. 

Maybe we should use an explicit image tag (e.g. `master-dba849b-pacific-centos-8`) if it makes more sense?

Reply:
we used testcontainers ryuk image from quay.io and we had CI stability problems related to that.
Docker Hub and GHCR both seem like better choices, unless things has substantially changed since then for the quay.io.

Reply:
Ryuk is unable to pull an image from quay.io so I've used the `master-dba849b-pacific-centos-8` tag (the latest tag from Docker Hub).